### ISSUE TITLE
Pointer arithmetic 

### ISSUE THREAD

**Issue by hannorein** [matthewholman/assist#11]
I get a couple of compiler warnings about the pointer arithmetic in `spk.c`. For examples:
```
spk.c:260:24: warning: arithmetic on a pointer to void is a GNU extension [-Wpointer-arith]
        val = (void *)pl->map + sizeof(double) * (pl->one[m][n] - 1)
```
I've fixed this in 860c72c6c0d89c9206948d47fc6ee62e60fa54a4 but you've changed it back. I'm wondering why. Did that not work?

**Comment by matthewholman** [comment]
Sorry for not commenting earlier.  That change caused the ephemeris-reading code to silently give incorrect results.  This is one for Rob to work on.

**Comment by hannorein** [comment]
Oh. I think I see what the issue was. That was my bad. I believe I've fixed it in #7d135a8. This doesn't change the calculation on my computer -- but please check.
